hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,63 @@
+RELEASE_TYPE: major
+
+Welcome to the next major version of Hypothesis!
+
+There are no new features here, as we release those in minor versions.
+Instead, 5.0 is a chance for us to remove deprecated features (many already
+converted into no-ops), and turn a variety of warnings into errors.
+
+If you were running on the last version of Hypothesis 4.x *without any
+Hypothesis deprecation warnings*, this will be a very boring upgrade.
+**In fact, nothing will change for you at all.**
+
+.. note::
+    This release drops support for Python 2, which has passed
+    `its end of life date <https://devguide.python.org/#status-of-python-branches>`__.
+    The `Python 3 Statement <https://python3statement.org/>`__ outlines our
+    reasons, and lists many other packages that have made the same decision.
+
+    ``pip install hypothesis`` should continue to give you the latest compatible version.
+    If you have somehow ended up with Hypothesis 5.0 on Python 2, you need to update your
+    packaging stack to ``pip >= 9.0`` and ``setuptools >= 24.2`` - see `here for details
+    <https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires>`__.
+
+
+Strategies
+~~~~~~~~~~
+- :func:`~hypothesis.strategies.integers` bounds must be equal to an integer,
+  though they can still be other types.
+- If :func:`~hypothesis.strategies.fractions` is passed a ``max_denominator``,
+  the bounds must have at most that denominator.
+- :func:`~hypothesis.strategies.floats` bounds must be exactly representable as a
+  floating-point number with the given ``width``.  If not, the error message
+  includes the nearest such number.
+- :func:`sampled_from([]) <hypothesis.strategies.sampled_from>` is now an error.
+- The values from the ``elements`` and ``fill`` strategies for
+  :func:`hypothesis.extra.numpy.arrays` must be losslessly representable in an
+  array of the given dtype.
+- The ``min_size`` and ``max_size`` arguments to all collection strategies must
+  be of type :class:`python:int` (or ``max_size`` may be ``None``).
+
+Miscellaneous
+~~~~~~~~~~~~~
+- The ``.example()`` method of strategies (intended for interactive
+  exploration) no longer takes a ``random`` argument.
+- You must pass either the ``target`` or ``targets`` argument to stateful rules,
+  but may not pass both.
+- :obj:`~hypothesis.settings.deadline` must be ``None`` (to disable), a
+  :class:`~python:datetime.timedelta`, or an integer or float number of milliseconds.
+- :obj:`~hypothesis.settings.derandomize` and :obj:`~hypothesis.settings.print_blob`
```

Comment:
Slightly weird phrasing that makes these sound more related than they are. Perhaps "Both of (...) must now be booleans, where previously they accepted other values."